corehq/apps/enterprise/static/enterprise/js/enterprise_dashboard.js

```diff
@@ -1,34 +1,224 @@
+            }
+
+            // the backend handles inclusive date ranges by adding a full day to the end date,
+            // so we'd like to minimize the results shown from the next day
+            return endDate.startOf('day');
+        });
+
+        self.startDate = ko.pureComputed(function () {
+            const presetType = self.presetType();
+            let startDate = null;
+            if (presetType === PRESET_LAST_30) {
+                startDate = self.endDate().clone().subtract(30, "days");
+            } else if (presetType === PRESET_PREV_MONTH) {
+                startDate = self.endDate().clone().startOf("month");
+            } else {
+                startDate = moment(self.customStartDate());
+            }
+
+            return startDate;
+        });
+
+        datePicker.subscribe("change.td", function () {
+            if (datePicker.dates.picked.length === 1) {
+                const selectedDate = moment(datePicker.dates.picked[0]);
+                datePicker.updateOptions({
+                    restrictions: {
+                        minDate: new tempusDominus.tempusDominus.DateTime(selectedDate.clone().subtract(maxDateRangeDays, 'days')),
+                        maxDate: new tempusDominus.tempusDominus.DateTime(selectedDate.clone().add(maxDateRangeDays, 'days')),
+                    },
+                });
+            } else {
+                datePicker.updateOptions({
+                    restrictions: {
+                        minDate: undefined,
+                        maxDate: undefined,
+                    },
+                });
+            }
+            self.customStartDate(datePicker.dates.picked[0]);
+            if (datePicker.dates.picked.length > 1) {
+                self.customEndDate(datePicker.dates.picked[1]);
+            } else {
+                self.customEndDate(datePicker.dates.picked[0]);
+            }
+        });
+
+        return self;
+    };
+
+    function updateDisplayTotal($element, kwargs) {
+        const $display = $element.find(".total");
+        const slug = $element.data("slug");
+        const requestParams = {
+            url: initialPageData.reverse("enterprise_dashboard_total", slug),
+            success: function (data) {
+                $display.html(Number(data.total).toLocaleString());
+            },
+            error: function (request) {
+                if (request.responseJSON) {
+                    alertUser.alert_user(request.responseJSON["message"], "danger");
+                } else {
+                    alertUser.alert_user(gettext("Error updating display total, please try again or report an issue if this persists."), "danger");
+                }
+                $display.html(gettext("??"));
+            },
+            data: kwargs,
+        };
+        $display.html('<i class="fa fa-spin fa-spinner"></i>');
+        $.ajax(requestParams);
+    }
+
     $(function () {
+        const datePicker = tempusDominus.createDateRangePicker(
```

Comment:
Given that this is using the default ` to ` separator, could it use `createDefaultDateRangePicker`, including adding the new start/end date behavior to that function? I'd like to deprecate `createDateRangePicker` eventually, or at least limit its use, rather than encourage people to pass in their own separators.

Reply:
Sure, I've gone ahead and moved it over. I did notice, however, that `getDateRangeSeparator` was not translating the string, so I applied `gettext` to it. I found it a bit sketchy that I think we're relying on the global `gettext`, so just wanted to make sure this change was safe for other usages of Tempus Dominus?

Reply:
Yeah, that's safe, it was an oversight that it wasn't used already. `gettext` should be safe to use everywhere, and in a requirejs context it shouldn't be directly accessing the global, since [this module](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/django.js) is [always included](https://github.com/dimagi/commcare-hq/blob/59685b1925f33ab9967d18765949a206a56b8c3e/corehq/apps/hqwebapp/templates/hqwebapp/partials/requirejs.html#L36), though I don't remember exactly why it's set up the way it is.